pkg/controller/node/nodecontroller_test.go

```diff
@@ -1265,37 +1267,43 @@ func TestMonitorNodeStatusEvictPodsWithDisruption(t *testing.T) {
 		if err := nodeController.monitorNodeStatus(); err != nil {
 			t.Errorf("%v: unexpected error: %v", item.description, err)
 		}
-		// Give some time for rate-limiter to reload
-		time.Sleep(500 * time.Millisecond)
-
 		for zone, state := range item.expectedFollowingStates {
 			if state != nodeController.zoneStates[zone] {
 				t.Errorf("%v: Unexpected zone state: %v: %v instead %v", item.description, zone, nodeController.zoneStates[zone], state)
 			}
 		}
-		zones := testutil.GetZones(fakeNodeHandler)
-		for _, zone := range zones {
-			// Time for rate-limiter reloading per node.
-			time.Sleep(50 * time.Millisecond)
-			nodeController.zonePodEvictor[zone].Try(func(value TimedValue) (bool, time.Duration) {
+		podEvicted = false
+		// Infinite loop, used for retrying in case ratelimiter fails to reload for Try function.
+		// this breaks when we have the status that we need for test case.
+		// TODO: Add some timeout for infinite loop to break.
+		for {
```

Comment:
If this needs to be 5 min. there's probably bug in the NC somewhere, which causes ratelimiter tokens to be used somewhere else. We need to understand why this fails instead of hacking around it to make it pass.